data/os/Debian.yaml

```
@@ -1,4 +1,5 @@
 ---
 unbound::pidfile: '/run/unbound.pid'
 unbound::runtime_dir: '/var/lib/unbound'
-unbound::purge_unbound_conf_d: true
+unbound::conf_d: "/etc/unbound/unbound.conf.d"
```

I thought i could use the following here but `$confdir` doesn't appear to be in scope
```yaml
unbound::conf_d: "%{confdir}/unbound.conf.d"
```

---

you can, but not with Puppet DSL interpolation, but with hiera interpolation:
```suggestion
unbound::conf_d: "%{lookup('confdir')}/unbound.conf.d"
```

---

If i do this then the spec test fails and when i look at the catalog the conf_d param is set to: ` File[/unbound.conf.d]` and we also see an untouched resource of `File[/unbound.conf.d]` in the coverage.  

Things i have tried

---

> i will accept this comment and let CI kick in case, just incase im going mad

---

oh indeed this isn't going to work. `confdir` is a puppet variable. I assumed it's a hiera value. in Hiera you can only reference hiera keys (because all puppet code is evaluated afterwards). So you would need to move the data from init.pp to hiera.